evaluate/task/task-transpile.go

```diff
@@ -170,49 +162,66 @@ func (t *TaskTranspile) Run(ctx evaltask.Context) (repositoryAssessment map[eval
+	originFilePathsWithLanguage = map[string]language.Language{}
 	packagePathAbsolute := filepath.Join(ctx.Repository.DataPath(), packagePath)
 
-	files, err := originLanguage.Files(fileLogger, filepath.Join(packagePathAbsolute, "implementation"))
+	files, err := os.ReadDir(filepath.Join(packagePathAbsolute, "implementation"))
 	if err != nil {
-		return "", "", pkgerrors.WithStack(err)
+		return nil, "", pkgerrors.WithStack(err)
+	}
+
+	for _, file := range files {
+		var originLanguage language.Language
+		for _, l := range language.Languages {
+			if filepath.Ext(file.Name()) == l.DefaultFileExtension() {
+				originLanguage = l
+			}
+		}
+		originFilePathsWithLanguage[filepath.Join(filepath.Join(packagePathAbsolute, "implementation", file.Name()))] = originLanguage
 	}
-	originFilePath = filepath.Join("implementation", files[0])
 
 	stubFilePath, err = packageSourceFile(fileLogger, packagePathAbsolute, ctx.Language)
 	if err != nil {
-		return "", "", err
+		return nil, "", err
 	}
 
-	return originFilePath, stubFilePath, nil
+	return originFilePathsWithLanguage, stubFilePath, nil
 }
 
 // validateTranspileRepository checks if the repository for the "transpile" task is well-formed.
 func validateTranspileRepository(logger *log.Logger, repositoryPath string, destinationLanguage language.Language) (err error) {
 	logger.Printf("validating repository %q", repositoryPath)
 
-	var originLanguage language.Language
-	if _, ok := destinationLanguage.(*golang.Language); ok {
-		originLanguage = &java.Language{}
-	} else {
-		originLanguage = &golang.Language{}
-	}
-
 	packagePaths, err := repositoryOnlyHasPackages(repositoryPath)
 	if err != nil {
 		return err
 	}
 
 	for _, packagePath := range packagePaths {
-		// Validate the implementation folder.
-		files, err := originLanguage.Files(logger, filepath.Join(packagePath, "implementation"))
+		implementationDirectoryPath := filepath.Join(packagePath, "implementation")
+		files, err := os.ReadDir(implementationDirectoryPath)
 		if err != nil {
 			return pkgerrors.WithStack(err)
-		} else if len(files) != 1 {
-			return pkgerrors.Errorf("package %q must have an \"implementation\" directory with just one %s source file to transpile", packagePath, originLanguage.Name())
-		} else if strings.HasSuffix(files[0], originLanguage.DefaultTestFileSuffix()) {
-			return pkgerrors.Errorf("package %q must have an \"implementation\" directory with only a %s source file, but found a test file %q", packagePath, originLanguage.Name(), files[0])
+		}
+
+		// Check if the implementation directory is well-formed.
+		for _, file := range files {
+			if file.IsDir() {
+				return pkgerrors.Errorf("the implementation directory %q must contain only source code files to transpile, but found one directory: %q", implementationDirectoryPath, file.Name())
+			}
+			var originLanguage language.Language
+			for _, language := range language.Languages {
```

Comment:
Additional validation check: for each language there can only be one implementation file.